### DISCUSSION TITLE
FOSS Hack 2024

### DISCUSSION THREAD

**Discussion by VishnuSanal** [TeamAmaze/AmazeFileManager#4195] [General]
# FOSS Hack 2024

Hi folks! We are thrilled to announce that Phase is partnering with [FOSS Hack](https://fossunited.org/fosshack/2024), an incredible event organized by FOSS United as part of their FOSS Hack Partner Projects Programme!

### Quick Links

- [Contribution Guidelines](https://github.com/TeamAmaze/AmazeFileManager/blob/release/4.0/CONTRIBUTING.md)
- [Code of Conduct](https://github.com/TeamAmaze/AmazeFileManager/blob/release/4.0/CODE_OF_CONDUCT.md)
- [Security Policy](https://github.com/TeamAmaze/AmazeFileManager/security/policy)
- [Telegram Group](https://t.me/AmazeFileManager)
- [good first issues](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aissue+is%3Aopen+label%3A%22Issue-Easy+%28good+first+issue%29%22) will be a good place to start to get familiarized with the code.

### Issues

There are three levels of issues.

- [`fosshack easy`](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aopen+is%3Aissue+label%3A%22fosshack+easy%22)
- [`fosshack medium`](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aopen+is%3Aissue+label%3A%22fosshack+medium%22)
- [`fosshack hard`](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aopen+is%3Aissue+label%3A%22fosshack+hard%22)

PS: issues labelled [`fosshack assigned`](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aopen+is%3Aissue+label%3A%22fosshack+assigned%22) are already assigned; refrain from picking them up. 

PPS: [quick link](https://github.com/TeamAmaze/AmazeFileManager/issues?q=is%3Aissue+is%3Aopen+label%3Afosshack%2A%2C%22fosshack+easy%22%2C%22fosshack+hard%22%2C%22fosshack+medium%22+-label%3A%22fosshack+assigned%22) to all available fosshack issues

PPPS! if you want to work on an issue without a fosshack label currently, feel free to let us know, we'll check & add 'em to the list if it is valid.

### Getting help

If you face any issues whilst setting up the project on your local machine, feel free to comment in this thread.

### FAQ

- regarding kotlin: we are not "migrating" the codebase to Kotlin per se; but, new files should be created in Kotlin, in case are.
- please use `release/4.0` as the base branch. also, point your PRs to the same branch.

**Comment by shebin-k-s** [comment]
we, team binary brains intent to work on the following issues as part of foss hackathon :
1. Allow to infinite scroll the tabs #2847
2. FTP Stops when the screen auto-sleeps #4125
3. Npe on searchview update result list in amaze file manager #4140 
4. When previewing files in an archive without a default "open with" application, cancelling to select an application will make every file unopenable #4205
5. Search in Recent Files Area-Search fosshack medium Issue-Feature #4179
6.Add direct camera integration #4055 
7.Batch uninstall apps feature request #777 
8.Custom location for app backup #3334

**Reply by VishnuSanal** [reply]
hi @shebin-k-s #4140 and #2847 is already being worked upon by someone -- we can't assign that to you folks. feel free to pick other issues. I have added `fosshack assigned` label to other issues & locked them for you. thanks for contributing to amaze, welcome aboard! :)

**Comment by aditya-ag26** [comment]
Hello,
We are Team Impala, and we would like to work on the following issues as part of the FOSS Hackathon:
1) [Documentation] Add screenshots to Readme #4063
2) Batch select in app manager #3093
3) Allow infinite scroll for tabs #2847
4) Searchable preferences #3178

Thank you for this opportunity.

**Comment by shebin-k-s** [comment]
We, Team Binary Brains, intend to work on the following additional issues as part of the FOSS hackathon:

1.Avoid unnecessary refresh on files and folders deletion (#3075).
2.Show folder sizes (#3783).

**Reply by VishnuSanal** [reply]
I'll assign #3075 right now. And, we are not sure about #3783, i.e, it is not yet decided right now tha whether we want that feature or not. (hence the `issue-discussion` label). We'll discuss that & let you know. Thanks again for your interest in contributing to Amaze!
